Add-ons/UmbracoCourier/Developer/DataResolvers.md

```diff
@@ -3,17 +3,196 @@
+
+### AscxFiles
+* **Full name:** `Umbraco.Courier.DataResolvers.ascxFiles`
+* **Triggers on:** Macros which have a .ascx file as its macro file
+* Collects resources for ascx files on macros, this can both be .dll or .cs files
+
+### ContentPicker
+* **Full name:** `Umbraco.Courier.DataResolvers.ContentPicker`
+* **Triggers on:** Propertydata, which have a contentpicker as data type (configured in courier.config)
+* If value is set, and is an int, courier will convert the value to the Node GUID, add the node as a dependency. On extraction the GUID will be converted back to the right ID.
+
+**Configuration**
+
+```xml
+<contentPickers>
+    <!-- add new datatype elements for data types that stores page ids (ex: "1242" or "1726,2362,2323") -->
+    <add key="contentPicker">158aa029-24ed-4948-939e-c3da209e5fba</add>
+    <add key="ultimatePicker">cdbf0b5d-5cb2-445f-bc12-fcaaec07cf2c</add>
+            
+    <add key="Ucomponents-XpathCheckboxlist">d2d46927-f4f8-4b1b-add7-661cc09a0539</add>
+    <add key="Ucomponents-XpathDropdownlist">57a62843-c488-4c29-8125-52f51873613e</add>
+    <add key="Ucomponents-AutoComplete">31aa0d5c-f8e1-4cdc-a66e-c7f8c09498ef</add>
+</contentPickers>
+```
+
+### CSSResources
+* **Full name:** `Umbraco.Courier.DataResolvers.CssResources`
+* **Triggers on:** Stylesheets
+* Includes images included in the stylesheet as resources
+
+### DampResolver
+* **Full name:** `Umbraco.Courier.DataResolvers.DampResolver`
+* **Triggers on:** Propertydata, which have a DAMP pick as data type
+* If value is set, and is an `int`, Courier will convert the value to the media GUID and add the media item as a dependency 
+* On extraction the GUID will be converted back to the right ID.
+
+### EmbeddedContent
+* **Full name:** `Umbraco.Courier.DataResolvers.EmbeddedContent`
+* **Triggers on:** Propertydata, which have a EmbeddedContent type as data type 
+* Replaces node IDs in the embedded content with corresponding GUIDs and converts them back again on extraction
+
+### Images
+* **Full name:** `Umbraco.Courier.DataResolvers.Images`
+* **Triggers on:** Propertydata, which contains an RTE 
+* Finds linked images in the RTE HTML and sorts out IDs, paths and resources
+
+
+### KeyValuePrevalueEditor
+* **Full name:** `Umbraco.Courier.DataResolvers.KeyValuePrevalueEditor`
+* **Triggers on:** Propertydata, which contains a keyvalue editor like dropdownlist, radiobutton list or checkboxlist 
+* Resolves prevalues from IDs to actual value, and back again on extraction
+
+**Configuration**
+
+```xml
+<keyValuePrevalueEditors>
+    <!-- Prevalue editors that store values as a key value pair in the built-in umbracp prevalue storage, identified by their full class-name -->
+    <add key="KeyValuePrevalueEditor">umbraco.editorControls.KeyValuePrevalueEditor</add>
+</keyValuePrevalueEditors>
+```
+
+### LocalLinks
+* **Full name:** `Umbraco.Courier.DataResolvers.LocalLinks`
+* **Triggers on:** Propertydata, which contains a the string {locallink: 
```

Comment:
```suggestion
* **Triggers on:** Propertydata, which contains the string {locallink: 
```
Further up it mentions the syntax as  [locallink:], might need to look further into this 😄